### DISCUSSION TITLE
Wagtail Work Hours

### DISCUSSION THREAD

**Discussion by Nigel2392** [wagtail/wagtail#11362] [Ideas]
### Is your proposal related to a problem?
Do you hate not getting paid enough? Perhaps you've forgotten to write down your worked hours? Wrote down too little?

### Describe the solution you'd like
Keep track of the time it took to edit an object!
It wouldn't be too hard to implement either I think. It seems like this can be built into wagtail's audit log (Reports, Site history).
We could for example; store the starting time in a session; and retrieve that when saving to know how long work was done on any given object reported in the site history.

### Describe alternatives you've considered
Could not find any wagtail packages for this; and it seems like this should be a core part of the CMS.

### Additional context
Saving when an object was created or edited is useful; but knowing how long it was edited for; and providing some nice user-based aggregates would be perfect for any employee and manager alike.

**Comment by thibaudcolas** [comment]
hi @Nigel2392, thanks for the suggestion. This seems like an interesting feature for some use cases but I’m not aware of any CMS having this as a core feature, and I don’t see a reason why you couldn’t build this as a package.

So I have converted this to GitHub Discussions – it seems interesting, but I see this needing more demonstration outside Wagtail itself first.

**Reply by Nigel2392** [reply]
Very well - I'll build it out as a package. 

Quick question; what do you think would be the preferred way to register models for keeping track of time spent editing?
1. Have all snippets/pages automatically be registered
2. Register models separately through the use of hooks

**Comment by SebCorbin** [comment]
This could be part of/developped along with (in a third-party package) auto-locking mechanism? https://github.com/wagtail/roadmap/issues/41
